### DISCUSSION TITLE
help me

### DISCUSSION THREAD

**Discussion by JenniClark** [community/community#110821] [Programming Help] [answered]
What is the best way to make an responsive design in HTML CSS?
First I do it in desktop design and when I want to do it in mobile design, it breaks everything, I'm stuck, help!
What do I do??

**Comment by thekarthikr** [comment]


**Reply by queenofcorgis** [reply]
@thekarthikr 

We’ve clarified our stance on using generative AI tools like ChatGPT within our Community via [this announcement](https://github.com/orgs/community/discussions/69006). Please review the guidelines to ensure your post meets them as failure to adhere to those rules can result in action taken by our moderator team.  You can read our updated [Code of Conduct](https://github.com/community/community/blob/baa5cd9020ef3f2bbf86dca1191d9c90f9b0822d/CODE_OF_CONDUCT.md#reasonable-use-of-ai-generated-content) and the [announcement](https://github.com/orgs/community/discussions/69006) for more details. Thank you for helping us maintain an authentic and beneficial space for everyone.

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @JenniClark ! We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Discussions category is solely related to conversations around the GitHub product Discussions.

We’ve moved your post to our Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.

**Comment by SRayen** [comment]
Media Queries:

Media queries are CSS statements that allow you to apply different styles based on specific screen size conditions. You can target devices based on width, height, orientation, and resolution.
Here's a basic example of a media query:

CSS
`@media only screen and (max-width: 768px) {
  /* Styles for screens less than 768px wide */
  .container {
    width: 80%;
  }
}`

**Comment by IsuruSankhajith** [comment]
Creating a responsive design in HTML and CSS involves using media queries, flexible layouts, and relative units. Start with the smallest screen size and gradually enhance for larger screens using media queries. Utilize percentages and viewport-relative lengths instead of fixed pixel values. Consider frameworks like Bootstrap or Flexbox for built-in responsive utilities. Test your design on different devices to ensure consistency.

**Comment by amankr1469** [comment]
Always use relative units and check what's the default display type of tag you are using. Making display flex and then playing elements in the web browser in Inspect elements can help you decide what you should choose.

**Comment by Randy0607** [comment]
Instead of using fixed pixel values for layout widths, use percentages. This allows elements to adjust their size relative to the screen width.

.container {
    width: 100%;
    max-width: 1200px; /* Optional: set a maximum width for larger screens */
    margin: 0 auto; /* Centers the container */
}

**Comment by shoyim** [comment]
https://chat.openai.com/share/eb1d34ef-2392-4a8c-b0e9-f3f219ff01d1

**Comment by ByDiamondGod** [comment, marked as answer]
Hi, I'll give you 3 tips to make a responsive design:

1 - If you already have the entire page in desktop format and want to make it mobile, start by checking where it breaks = (breakpoint) for example at 768px width.

2 - Once you know where it breaks, start commenting out the HTML or using display: none for the sections that you're not going to make responsive now. And so, you do it in parts, start with the header, then the main, sections, asides, and finally the footer.

Tip: I tell you this because there are occasions when you're doing the responsive design well, but if you have other tags below the page, it may cause the page to break and not look good, even if you did it correctly. That's why I advise you to use display: none with CSS in the sections that you haven't prepared for responsive mode yet.

3 - I'll leave you with this reminder about media

// MAX (From here, up to -px)
// MIN (From here, up to +px)

media screen and (m?-width: ?px) {

}

Feel free to ask if you need any further assistance!

ESP

Te doy a dar 3 tips para poder hacer un diseño adaptable

1 - Si ya tienes toda la página en formato desktop y quieres pasarlo a mobile, empieza verificando en donde se rompe = (quiebre - breakpoint) ejemplo a los 768px de width.

2 - Cuando ya sabes donde se rompe, empezá a comentar el html o hacer un display: none de las secciones que por el momento no vas a hacerle el responsive. Y así lo vas haciendo por partes. Ejemplo: el header, luego el main, secciones, asides y por el ultimo el footer.
Consejo: Te digo esto porque hay ocasiones que estás haciendo bien el diseño adaptable, pero si tenes otras etiquetas abajo de la página, puede ser que esto haga que rompa la página y no se vea bien, aunque si lo hayas hecho bien. Por eso te aconsejo hacer un display: none con CSS en las secciones que todavía no le preparaste el modo responsive

3 - Te dejo este recordatorio de media

// MAX (Desde aca, haste - px)
// MIN (Desde aca, hasta + px)

media screen and (m-width: px) {

}

Espero que te sirva, éxitos. ¡Vos podes!

**Reply by JenniClark** [reply]
You are the best. You really have experience in Frontend.
Nice comment and examples.
